bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/nano/WlanThermoNanoV1CommandHandler.java

```diff
@@ -0,0 +1,266 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wlanthermo.internal.api.nano;
+
+import static org.openhab.binding.wlanthermo.internal.WlanThermoBindingConstants.*;
+
+import java.awt.*;
+import java.math.BigInteger;
+import java.util.List;
+
+import javax.measure.Unit;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.wlanthermo.internal.WlanThermoInputException;
+import org.openhab.binding.wlanthermo.internal.WlanThermoUnknownChannelException;
+import org.openhab.binding.wlanthermo.internal.api.nano.dto.data.Channel;
+import org.openhab.binding.wlanthermo.internal.api.nano.dto.data.Data;
+import org.openhab.binding.wlanthermo.internal.api.nano.dto.data.Pm;
+import org.openhab.binding.wlanthermo.internal.api.nano.dto.data.System;
+import org.openhab.binding.wlanthermo.internal.api.nano.dto.settings.Settings;
+import org.openhab.core.library.types.*;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link WlanThermoNanoV1CommandHandler} is responsible for mapping the Commands to the respective data fields
+ * of the API.
+ *
+ * @author Christian Schlipp - Initial contribution
+ */
+@NonNullByDefault
+public class WlanThermoNanoV1CommandHandler {
+
+    public static State getState(ChannelUID channelUID, @Nullable Data data, @Nullable Settings settings)
+            throws WlanThermoUnknownChannelException, WlanThermoInputException {
+        if (channelUID.getGroupId() == null || data == null || settings == null) {
+            throw new WlanThermoInputException();
+        }
+
+        String groupId = channelUID.getGroupId();
+        System system = data.getSystem();
+        Unit<Temperature> unit = "F".equals(system.getUnit()) ? ImperialUnits.FAHRENHEIT : SIUnits.CELSIUS;
+        List<Channel> channelList = data.getChannel();
+
+        if (SYSTEM.equals(groupId)) {
+            switch (channelUID.getIdWithoutGroup()) {
+                case SYSTEM_SOC:
+                    return new DecimalType(system.getSoc());
+                case SYSTEM_CHARGE:
+                    return OnOffType.from(system.getCharge());
+                case SYSTEM_RSSI_SIGNALSTRENGTH:
+                    int dbm = system.getRssi();
+                    if (dbm >= -80) {
+                        return SIGNAL_STRENGTH_4;
+                    } else if (dbm >= -95) {
+                        return SIGNAL_STRENGTH_3;
+                    } else if (dbm >= -105) {
+                        return SIGNAL_STRENGTH_2;
+                    } else {
+                        return SIGNAL_STRENGTH_1;
+                    }
```

Comment:
Where did you get these numbers from? To my knowledge a dbm of -95 is pretty much the noise floor for most wireless applications.

Reply:
As I couldn't find any generic conversion function, I've checked the source code of the WlanThermo device and took the same numbers. 
Benefit: The signal indicator now gives the same number of bars as shown on the display of the device.

Reply:
Well if it matches what is shown on the device then I guess that is the correct implementation. Keep it as is then.